docs/en/02_Developer_Guides/14_Files/01_File_Management.md

```diff
@@ -2,9 +2,122 @@ summary: Learn how to work with File and Image records
 
 # File Management
 
-## Files, Images and Folders as database records
+## Storage via database columns
 
-All files, images and folders in the 'assets' directory are stored in the database. Each record has the following database fields:
+Asset storage is provided out of the box via a [Flysystem](http://flysystem.thephpleague.com/) backed store.
+However, any class that implements the `AssetStore` interface could be substituted to provide storage backends
+via other mechanisms.
+
+Internally, files are stored as `[api:DBFile]` records on the rows of parent objects. These records are composite fields
+which contain sufficient information useful to the configured asset backend in order to store, manage, and 
+publish files. By default this composite field behind this field stores the following details:
+
+
+| Field name     | Description |
+| ----------     | -----------   
+| `Hash`         | The sha1 of the file content, useful for versioning (if supported by the backend) |
+| `Filename`     | The internal identifier for this file, which may contain a directory path (not including assets). Multiple versions of the same file will have the same filename. |
+| `Variant`      | The variant for this file. If a file has multiple derived versions (such as resized files or reformatted documents) then you can point to one of the variants here. |
+
+
+Note that the `Hash` and `Filename` always point to the original file, if a `Variant` is specified. It is up to the
+storage backend to determine how variants are managed.
+
+## Compatibility with 3.x filename paths
+
+If upgrading to 4.0 from earlier versions when using the default asset store, it's important to consider
+how existing files will be migrated.
+
+Because the filesystem now uses the sha1 of file contents in order to version multiple versions under the same
+filename, the default storage paths in 4.0 will not be the same as in 3.
+
+In order to retain existing file paths in line with framework version 3 you should set the
+`\SilverStripe\Filesystem\Flysystem\FlysystemAssetStore.legacy_paths` config to true.
+Note that this will not allow you to utilise certain file versioning features in 4.0.
+
+	:::yaml
+	\SilverStripe\Filesystem\Flysystem\FlysystemAssetStore:
+		legacy_paths: true
+
+## Loading content into `DBFile`
+
+A file can be written to the backend from a while which exists on the local filesystem (but not necessarily
+within the assets folder).
+
+For example, to load a temporary file into a DataObject you could use the below:
+
+
+	:::php
+	<?
+	class Banner extends DataObject {
+		private static $db = array(
+			'Image' => 'DBFile'
+		);
+	}
+
+	// Image could be assigned in other parts of the code using the below
+	$banner = new Banner();
+	$banner->Image->setFromLocalFile($tempfile['path'], 'uploads/banner-file.jpg');
```

Comment:
It doesn't need to be anymore. :) Composite DBFields can be accesed via normal getters now.